ttnn/cpp/ttnn/operations/pool/rotate/device/kernels/dataflow/reader_rotate_nearest_interleaved.cpp

```diff
@@ -0,0 +1,96 @@
+    const auto input_tensor_accessor = TensorAccessor(src_args, input_addr, input_stick_nbytes);
+    constexpr uint32_t batch_size = num_cb_pages / 2 < 5 ? num_cb_pages / 2 : 5;
+
+    uint32_t fill_stick_addr;
+    if constexpr (fill_is_zero) {
+        fill_stick_addr = MEM_ZEROS_BASE;
+    } else {
+        fill_stick_addr = get_write_ptr(fill_cb_index);
+        volatile tt_l1_ptr uint16_t* fill_ptr = reinterpret_cast<volatile tt_l1_ptr uint16_t*>(fill_stick_addr);
+        for (uint32_t c = 0; c < input_channels; c++) {
+            fill_ptr[c] = static_cast<uint16_t>(fill_value_bf16);
+        }
+    }
+
+    for (uint32_t local_stick_idx = 0; local_stick_idx < num_sticks;) {
+        uint32_t sticks_this_batch =
+            (num_sticks - local_stick_idx) < batch_size ? (num_sticks - local_stick_idx) : batch_size;
+        cb_reserve_back(output_cb_index, sticks_this_batch);
+        uint32_t l1_write_addr = get_write_ptr(output_cb_index);
+
+        for (uint32_t i = 0; i < sticks_this_batch; i++, local_stick_idx++) {
+            const uint32_t global_stick_idx = start_stick_id + local_stick_idx;
+
+            const uint32_t batch_idx = global_stick_idx / (input_height * input_width);
+            const uint32_t spatial_idx = global_stick_idx % (input_height * input_width);
+            const uint32_t y_out = spatial_idx / input_width;
+            const uint32_t x_out = spatial_idx % input_width;
+
+            const fixed_point_t x_out_q16 = int_to_q16(x_out);
+            const fixed_point_t y_out_q16 = int_to_q16(y_out);
+            const fixed_point_t x_centered = q16_sub(x_out_q16, center_x);
+            const fixed_point_t y_centered = q16_sub(y_out_q16, center_y);
+
+            const fixed_point_t x_in = q16_mul_sub_add(x_centered, cos_angle, y_centered, sin_angle, center_x);
+            const fixed_point_t y_in = q16_mul_add_add(x_centered, sin_angle, y_centered, cos_angle, center_y);
+
+            const int32_t nearest_x = q16_to_int_round(x_in);
+            const int32_t nearest_y = q16_to_int_round(y_in);
+
+            const bool x_valid = is_coordinate_valid(nearest_x, input_width);
+            const bool y_valid = is_coordinate_valid(nearest_y, input_height);
+
+            if (x_valid && y_valid) {
+                const uint32_t input_stick_index =
+                    batch_idx * (input_height * input_width) + nearest_y * input_width + nearest_x;
+                const uint64_t input_noc_addr = input_tensor_accessor.get_noc_addr(input_stick_index);
+                noc_async_read(input_noc_addr, l1_write_addr, input_stick_nbytes);
+            } else {
+                uint32_t copy_size = input_stick_nbytes_unaligned;
+                if constexpr (fill_is_zero) {
+                    if (copy_size > MEM_ZEROS_SIZE) {
+                        copy_size = MEM_ZEROS_SIZE;
+                    }
+                }
+                noc_async_read(get_noc_addr(fill_stick_addr), l1_write_addr, copy_size);
```

Comment:
in case of zero padding you cant do a single noc trasacation for every stick size as amount of zeros starting at MEM_ZEROS_BASE is limited to MEM_ZEROS_SIZE, if your stick is bigger than that you need multipel transactions.
Better way would be to prepare this fill cb in advance (with potentially multipel transactions), and than use single transcation here for the target